tests/ui/ui_tester.py

```diff
@@ -0,0 +1,173 @@
+import argparse
+import os
+import time
+
+from abc import ABC, abstractmethod
+from diffimg import diff
+from selenium import webdriver
+from selenium.webdriver import ChromeOptions, FirefoxOptions
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+class UITester(ABC):
+    _SCREENSHOT_DIFF_THRESHOLD_PERCENT = 9
+    _BASE_PATH = os.path.dirname(os.path.abspath(__file__))
+
+    def __init__(self, test_name, base_url='http://localhost'):
+        self._test_name = test_name
+        self._base_url = base_url
+
+    @abstractmethod
+    def test(self):
+        pass
+
+    def run(self):
+        def add_headless(browser_options):
+            if args.headless:
+                browser_options.add_argument('--headless')
+
+        # Test Firefox
+        options = FirefoxOptions()
+        add_headless(options)
+        self._driver = webdriver.Firefox(log_path='', firefox_options=options)
+        self.test()
+        self._driver.close()
+
+        # Test Chrome
+        options = ChromeOptions()
+        add_headless(options)
+        self._driver = webdriver.Chrome(chrome_options=options)
+        self.test()
+        self._driver.close()
+
+    def login(self, username='codalab', password='codalab'):
```

Comment:
I think we define the default username and password somewhere else in another file -- could we import those here?

Reply:
Do you know which file I can look into? Also, I left these as parameters we can pass in, just in case, we wanted to test creating a new user account.

Reply:
It appears they are hardcoded here: https://github.com/codalab/codalab-worksheets/blob/5dfbacf6a544403cfb4f399a5b01657ba28b7870/.travis.yml#L21

Perhaps you could set the default to `os.getenv("CODALAB_USERNAME")` and `os.getenv("CODALAB_PASSWORD")`.